src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -169,36 +172,49 @@ class JesBackend extends Backend with LazyLogging {
     // Not possible to currently get stdout/stderr so redirect everything and hope the WDL isn't doing that too
     val redirectedCommand = s"$instantiatedCommandLine > $LocalStdoutValue 2> $LocalStderrValue"
 
-    val status = Pipeline(redirectedCommand, workflowDescriptor, call, jesParameters, GoogleProject, JesConnection).run.waitUntilComplete()
+    val run = Pipeline(redirectedCommand, workflowDescriptor, call, jesParameters, GoogleProject, JesConnection).run
+    cancelButtonRegistrationFunction apply { () => run.abort() }
+    try {
+      val status = run.waitUntilComplete()
 
-    def taskOutputToRawValue(taskOutput: TaskOutput): Try[WdlValue] = {
-      val jesOutputFile = unsafeJesOutputs find {_.name == taskOutput.name} map {j => WdlFile(j.gcs)}
-      jesOutputFile.map(Success(_)).getOrElse(taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions))
-    }
-
-    val outputMappings = call.task.outputs.map { taskOutput =>
-      val rawValue = taskOutputToRawValue(taskOutput)
-      logger.debug(s"JesBackend setting ${taskOutput.name} to $rawValue")
-      taskOutput.name -> rawValue
-    }.toMap
-
-    val warnAboutStderrLength: BigInteger =
-      if (!call.failOnStderr) BigInteger.valueOf(0)
-      else JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(callGcsPath).gcs))
+      def taskOutputToRawValue(taskOutput: TaskOutput): Try[WdlValue] = {
+        val jesOutputFile = unsafeJesOutputs find {
+          _.name == taskOutput.name
+        } map { j => WdlFile(j.gcs) }
+        jesOutputFile.map(Success(_)).getOrElse(taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions))
+      }
 
-    if (warnAboutStderrLength.intValue > 0) {
-      Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: stderr has length $warnAboutStderrLength for command: $instantiatedCommandLine"))
-    } else {
-      status match {
-        //
-        case Run.Success(created, started, finished) =>
-          unwrapOutputValues(outputMappings, workflowDescriptor)
-        case Run.Failed(created, started, finished, errorCode, errorMessage) =>
-          Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: errorCode $errorCode for command: $instantiatedCommandLine. Message: $errorMessage"))
+      val outputMappings = call.task.outputs.map { taskOutput =>
+        val rawValue = taskOutputToRawValue(taskOutput)
+        logger.debug(s"JesBackend setting ${taskOutput.name} to $rawValue")
+        taskOutput.name -> rawValue
+      }.toMap
+
+      val warnAboutStderrLength: BigInteger =
+        if (!call.failOnStderr) BigInteger.valueOf(0)
+        else JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(callGcsPath).gcs))
+
+      if (warnAboutStderrLength.intValue > 0) {
```

Comment:
This is actually from a previous commit. I think the awkward juggling is to avoid fetching stderr from GCS if you don't need to, avoid fetching it twice since you only need to find its length once, and avoid having too many nested ifs.